### PR TITLE
[ZEPPELIN-5160]. Improve github action CI

### PR DESCRIPTION
**Pull request by zjffdu**
### What is this PR for?

This PR improve the github action CI
* Disable unstable test BasePythonInterpreterTest#testRedefinitionZeppelinContext
* Fix the failure of flink 1.10, this is due to python package issue.

The github action still fails sometimes, because of network issue or other weird issue, we have to rerun it again.

### What type of PR is it?
[ Improvement]

### Todos
* [ ] - Task

### What is the Jira issue?
* https://issues.apache.org/jira/browse/ZEPPELIN-5160

### How should this be tested?
* CI pass

### Questions:
* Does the licenses files need update? NO
* Is there breaking changes for older versions? No
* Does this needs documentation? NO

### CODE REVIEW

**Comment by zjffdu** [comment]
@Reamer Could you help review it ?

**Review by Reamer — COMMENTED** [review]
I see that you prefer pip over conda. Interesting blog showing the differences: https://www.anaconda.com/blog/understanding-conda-and-pip
I prefer conda because it manages an isolated environment with system libraries, python, R binaries.
I know pipy has more packages than conda. When an environment is installed with conda, conda validates that all the environment dependencies work together.


I also see that you prefer fixed version of dependencies. I would prefer more open dependencies at least for apache-flink 1.11.x because we should check if zeppelin works with newly installed environments. I don't know any user who installs packages with a fixed version.
In my opinion, a fixed version makes sense when we check Zeppelin against a dependency that is no longer being developed. For example apache-flink 1.10.x or apache-spark 1.x

File: python/src/test/java/org/apache/zeppelin/python/BasePythonInterpreterTest.java [hunk 1]
```diff
@@ -354,22 +354,48 @@ public void testZeppelinContext() throws InterpreterException, InterruptedExcept
     assertEquals("%text world\n", context.out.getCurrentOutput().toString());
   }
 
-  @Test
+  //@Test
```
[the remark this answers is not in the document]
Comment (zjffdu) [reply python/src/test/java/org/apache/zeppelin/python/BasePythonInterpreterTest.java]: Fixed

**Comment by zjffdu** [comment]
Thanks for the review @Reamer Regarding the conda vs pip, I think here pip would also install python packages in the conda env, because the pip is installed via conda as well.

**Comment by Reamer** [comment]
Yes, that's true, but in general we should avoid using pip in a Conda environment  - [Using Pip in a Conda Environment](https://www.anaconda.com/blog/using-pip-in-a-conda-environment)

It's good, that you could fix the apache-flink 1.10.x tests. 

PullRequest LGTM

**Pull request closed without merging**